tools/c7n_azure/c7n_azure/resources/network_interface.py

```diff
@@ -29,3 +34,52 @@ class resource_type(object):
             'location',
             'resourceGroup'
         )
+
+
+@NetworkInterface.filter_registry.register('effective-route-table')
+class EffectiveRouteTableFilter(ValueFilter):
+    schema = type_schema('effective-route-table', rinherit=ValueFilter.schema)
+
+    def process(self, resources, event=None):
+        futures = []
+        results = []
+        chunk_size = 20
+
+        # Process each resource in a separate thread, returning all that pass filter
+        with self.executor_factory(max_workers=3) as w:

```

Comment:
Use the ThreadHelper https://github.com/cloud-custodian/cloud-custodian/blob/9af4fe8e1bb83b525c12ddb2c3d83c0055e87c23/tools/c7n_azure/c7n_azure/utils.py#L142